src/CommandingOfficers/AW1/OS/Max.java

```diff
@@ -0,0 +1,115 @@
+package CommandingOfficers.AW1.OS;
+
+import java.util.ArrayList;
+
+import CommandingOfficers.*;
+import CommandingOfficers.AW1.AW1Commander;
+import Engine.GameScenario;
+import Engine.Combat.StrikeParams;
+import Engine.Combat.StrikeParams.BattleParams;
+import Engine.UnitMods.DirectDamageModifier;
+import Engine.UnitMods.UnitModifier;
+import Engine.UnitMods.UnitMovementModifier;
+import Engine.UnitMods.UnitTypeFilter;
+import Terrain.MapMaster;
+import UI.UIUtils;
+import Units.UnitContext;
+import Units.UnitModel;
+
+public class Max extends AW1Commander
+{
+  private static final long serialVersionUID = 1L;
+
+  private static final CommanderInfo coInfo = new instantiator();
+  public static CommanderInfo getInfo()
+  {
+    return coInfo;
+  }
+  private static class instantiator extends CommanderInfo
+  {
+    private static final long serialVersionUID = 1L;
+    public instantiator()
+    {
+      super("Max_1", UIUtils.SourceGames.AW1, UIUtils.OS);
+      infoPages.add(new InfoPage(
+            "Max (AW1)\n"
+          + "Dependable and brave. Over-protective of Sami and Andy.\n"
+          + "Direct combat units have high firepower. Distance units are weak and have small attack ranges.\n"
+          + "(+50/0 vehicle direct combat, indirects -1 range and 0.9x/1.1x damage dealt/taken.)\n"));
+      infoPages.add(new InfoPage(new MaxForce(null),
+            "Increases all abilities of direct combat units.\n"
+          + "(+20/0 (187/110) vehicle direct combat, +1 move to non-indirect vehicles)\n"
+          + "1.1x/0.9x damage dealt/taken.\n"));
+      infoPages.add(new InfoPage(
+            "Hit: Weight training\n"
+          + "Miss: Studying"));
+      infoPages.add(AW1_MECHANICS_BLURB);
+    }
+    @Override
+    public Commander create(GameScenario.GameRules rules)
+    {
+      return new Max(rules);
+    }
+  }
+
+  public Max(GameScenario.GameRules rules)
+  {
+    super(coInfo, rules);
+
+    addCommanderAbility(new MaxForce(this));
+  }
+
+  @Override
+  public void modifyUnitAttack(StrikeParams params)
+  {
+    if( params.battleRange < 2 )
+      params.attackPower += 50;
+    if( params.attacker.model.isAny(UnitModel.INDIRECT) )
+    {
+      params.attackerDamageMultiplier *=  90;
+      params.attackerDamageMultiplier /= 100;
+    }
+  }
```

Comment:
Is there a reason to prefer these modify functions over just attaching a permanent `UnitModifier` to the Commander? Looks like we still use that approach in other places.

Reply:
The Commander _is_ a `UnitModifier`.
But like, other than that? Fewer object types.

Reply:
To elaborate a bit:
In places where we want a temporary effect, applying a temporary `UnitModifier` makes sense (since ideally it should work across COs for my jank tag mechanics).

Very basic code for those is ideal, since then you can reuse them.
But oh no, I want to apply the bonus only to some units!
A role filter can usually do the job. It's a bit more abstract, but it makes modifiers composable in nice ways.

As for the direct attack modifier itself, it gets a pass because it will have 4 "unique" users with each one using it more than once.


I guess I can see having Max just use a permanent `UnitModifier` for his direct attack buff.
That creates an arguably unnecessary object, isn't much less code than overriding a method, and he'd still need code for his debuff.
So I feel like standardizing COs on custom code for their D2Ds makes the most sense.

Reply:
> As for the direct attack modifier itself, it gets a pass

Not sure how that breaks your rules - just because it does its own filtering?

Otherwise fair enough. Potato potato, I was just curious.

Reply:
Yeah, it does its own filtering so it's not composable... but the set of buffs that make sense for direct combat is like 3, so it'd be a much more minimal filter class.
I guess if we add DS skills we could break it apart, but I really don't plan to do that.

Reply:
> Yeah, it does its own filtering so it's not composable

Dumb question then - why do that instead of composing it with the dedicated filter class?

Reply:
Because the dedicated filter class can't filter by "at range 1", and I want to use my own silly interpretation of how Max works.

Reply:
ooooookay

Reply:
I mean, it's the same, right? I can't think of when these wouldn't be the same. But okay.

Reply:
Anti-Tanks.
I am a very silly goose.